jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -79,6 +99,11 @@ public static DockerClient newClient(Path dockerExecutable) {
     this.processBuilderFactory = processBuilderFactory;
   }
 
+  @VisibleForTesting
+  Function<List<String>, ProcessBuilder> getProcessBuilderFactory() {
```

Comment:
nit: non-public methods go below public in code.

BTW, instead of having this new getter, is it possible to make `defaultProcessBuilderFactory()` `@VisibleForTesting`?

Reply:
Yes, but I do not think that this is a good solution. 

We want to test the actual use case, which is creating the process builder using one of the defined `newClient(*)` methods. Then, however, the only way to access the private process builder object is using the `getProcessBuilderFactory` getter.

If we are going to create the object directly using `defaultProcessBuilderFactory()`, we will not test the whole execution path.

Reply:
What you said makes sense, but I think this is basically a matter of the scope of a unit test, like a small unit test vs. medium or large unit test (more toward an end-to-end integration test). It's OK and meaningful to have narrowly scoped unit tests that test only one method, and if it's possible, it's also meaningful to have another larger tests that cover a  longer execution path at a higher level. I just feel it doesn't seem worth adding a new method (especially a simple getter) just to enable a test. After all, what you get from the getter is just an object built once from `defaultProcessBuilderFactory` at construction.

Reply:
To add, I'm trying to avoid spreading the practice like, "oh, I cannot access these fields in tests, but I want to directly examine the values in tests. I'll just make them visible by adding getters." It feels like, you may be more focused on testing internal implementation or design that is meant to be encapsulated.

Reply:
I can use `defaultProcessBuilderFactory` but then the last assertion in `DockerDaemonImageTest` is no longer feasible.